examples/with-tuist/tuist-workspace/README.md

```diff
@@ -0,0 +1,44 @@
+# Tuist iOS Example
+
+This is an example project demonstrating how to use the `cargo-xcframework` plugin with a Tuist generated iOS project.
+
+## Getting Started
+
+1. **Install Tuist**: If you haven't already, first [install Tuist](https://docs.tuist.io/guide/introduction/installation.html#installation).
+2. **Build and Run**: Use the `tuist run TuistIos` command, just like `cargo run`. Cool, right?
+
+## Static library
+
+First, build the XCFramework:
+
+```bash
+# In the root of the repo:
+cargo run -- --manifest-path examples/with-tuist/mymath-lib/Cargo.toml --lib-type staticlib
```

Comment:
When running this, I get:
```
✅ Run create xcframework success, output:
examples/with-tuist/mymath-lib/target/xcframework/MyMath.xcframework
failed to rename file from examples/with-tuist/mymath-lib/target/xcframework/MyMath.xcframework to examples/with-tuist/mymath-lib/target/MyMath.xcframework
```

- You probably need to remove the destination folder, if it exists, before the rename
- Feels a bit wrong that this is reported as success

Reply:
In some way, this is a known issue at this moment.
-  `create_xcframework` reports success.
-  The failure to rename is the existing behavior during "zip or move" final step:
```rust
        let to = conf.target_dir.join(format!("{module_name}.xcframework"));
        fs_err::rename(from, &to)?;
```
Since this PR focuses on extracting and replacing the core logic, I haven't addressed it yet. I will quickly fix it by checking existence. 

Additionally, in the next coming PR, I will enhance the `build` entry for CLI and user custom cases and devote more time to checking and improving UX issues.